doc/asab/storage.rst

```diff
@@ -0,0 +1,309 @@
+.. currentmodule:: asab.storage
+
+Storage
+=======
+
+ASAB provides a service for storing data. Data can be stored in memory or in dedicated document database. For now, `MongoDB <https://www.mongodb.com/>`_ and `Elastic Search <https://www.elastic.co/>`_ databases are supported.
+
+Specification of the storage type
+---------------------------------
+
+In order to use `asab.storage`, first you have to specify the type of storage. You can add configurations in the config file:
+
+.. code:: ini
+
+    [asab:storage]
+    type=mongodb
+
+or you can set it manually in the ASAB app:
+
+.. code:: python
+
+    import asab
+    import asab.storage
+
+    asab.Config.add_defaults(
+        {
+            'asab:storage': {
+                'type': 'mongodb'
+            }
+        }
+    )
+
+The options for the storage type are:
+
+- `inmemory`: Collects data directly in memory
+- `mongodb`: Collects data using MongoDB database. Depends on `pymongo <https://pymongo.readthedocs.io/en/stable/>`_ and `motor <https://motor.readthedocs.io/en/stable/api-asyncio/asyncio_motor_collection.html>`_ libraries.
+- `elasticsearch`: Collects data using Elastic Search database. Depends on `aiohttp <https://docs.aiohttp.org/en/latest/>`_ library.
+
+Although these three databases are different, accessing the database and manipulation with collections is done by using the same methods.
+
+For accessing the storage, simply `add asab.storage.Module`` when initializing and register the service.
```

Comment:
It is subjective what is "simple". But it is ok here, we don't need to be 100% official.